server/datastore/mysql/migrations/tables/20240821160025_AddRemovedToInstalls.go

```diff
@@ -0,0 +1,103 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240821160025, Down_20240821160025)
+}
+
+func Up_20240821160025(tx *sql.Tx) error {
+	if !columnExists(tx, "host_software_installs", "removed") {
+		if _, err := tx.Exec("ALTER TABLE host_software_installs ADD COLUMN removed TINYINT NOT NULL DEFAULT 0"); err != nil {
+			return fmt.Errorf("failed to add removed to host_software_installs: %w", err)
+		}
+	}
+
+	if !columnExists(tx, "host_vpp_software_installs", "removed") {
+		if _, err := tx.Exec("ALTER TABLE host_vpp_software_installs ADD COLUMN removed TINYINT NOT NULL DEFAULT 0"); err != nil {
+			return fmt.Errorf("failed to add removed to host_vpp_software_installs: %w", err)
+		}
+	}
+
+	// Mark software installs as removed if the software is no longer installed on the host.
+	// Note that some software never shows up in software table after being installed (because software detail query can't find it).
```

Comment:
What's an example? Does this happen when package title does not match installed software title?

Reply:
Just a thought: Maybe not worth doing this part if it may fail for some hosts (and better to just document this change is fixed for new host installs)

Reply:
Yes, we will leave alone any installed software where the package title does not match the installed software title.